core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -3465,8 +3511,32 @@ private Name() {} // prevent instantiation
         "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.watermark\\.low\\.ratio"),
     ;
 
+    // puts property creators in a nested class to avoid NPE in enum static initialization
+    private static class PropertyCreators {
+      private static final BiFunction<String, PropertyKey, PropertyKey> DEFAULT_PROPERTY_CREATOR =
+          createPropertyCreator(Scope.ALL, ConsistencyCheckLevel.IGNORE);
+      private static final BiFunction<String, PropertyKey, PropertyKey> UFS_PROPERTY_CREATOR =
+          createPropertyCreator(Scope.SERVER, ConsistencyCheckLevel.ENFORCE);
+      private static final BiFunction<String, PropertyKey, PropertyKey> JOURNAL_PROPERTY_CREATOR =
+          createPropertyCreator(Scope.MASTER, ConsistencyCheckLevel.ENFORCE);
+
+      private static BiFunction<String, PropertyKey, PropertyKey> createPropertyCreator(Scope scope,
+          ConsistencyCheckLevel consistencyCheckLevel) {
+        return (name, baseProperty) -> new Builder(name)
+            .setDisplayType(baseProperty.getDisplayType())
+            .setDefaultSupplier(baseProperty.getDefaultSupplier())
+            .setScope(scope)
+            .setConsistencyCheckLevel(consistencyCheckLevel)
+            .build();
+      }
+    }
+
+    private static final String NESTED_GROUP = "nested";
     private final String mFormat;
     private final Pattern mPattern;
+    private final Builder mPropertyBuilder;
+    private BiFunction<String, PropertyKey, PropertyKey> mPropertyCreator =
```

Comment:
with both a property builder and a property creator, it isn't very clear which one should take precedence. Could we remove the property builder and make the `PropertyKey` argument for property key creators `Nullable`? Then we could have a `DEFAULT_NON_NESTED_PROPERTY_CREATOR` as well as a `DEFAULT_NESTED_PROPERTY_CREATOR`, and only deal with builders inside `PropertyCreators`

Reply:
Refactored to unify the builder and the creator.